runtime/tests/checker/resources_test.go

```diff
@@ -4387,5 +4387,96 @@ func TestCheckInvalidResourceLossInNestedContractResource(t *testing.T) {
 	errs := ExpectCheckerErrors(t, err, 1)
 
 	assert.IsType(t, &sema.ResourceLossError{}, errs[0])
+}
+
+// https://github.com/onflow/cadence/issues/73
+//
+func TestCheckResourceMoveMemberInvocation(t *testing.T) {
+
+	t.Parallel()
+
+	t.Run("invalid use as argument", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+
+          resource Test {
+              fun use(_ test: @Test) {
+                  destroy test
+              }
+          }
+
+          fun test() {
+              let test <- create Test()
+              test.use(<-test)
+          }
+        `)
+
+		errs := ExpectCheckerErrors(t, err, 1)
+
+		assert.IsType(t, &sema.ResourceUseAfterInvalidationError{}, errs[0])
+	})
+
+	t.Run("valid use", func(t *testing.T) {
```

Comment:
Would perhaps be good to have a test case that includes a Resource method that accepts a Resource of the same type, and is invoked with a non `self` instance.

Reply:
Not quite sure if it's what you suggested, but added another test case in 94c7a3b75351b7b15589b423563e8c768c7fc877